operator/pkg/reconciliation/reconcile_racks.go

```diff
@@ -676,52 +676,84 @@ func (rc *ReconciliationContext) CheckRackPodLabels() result.ReconcileResult {
 	return result.Continue()
 }
 
-func shouldUpsertSuperUser(dc api.CassandraDatacenter) bool {
-	lastCreated := dc.Status.SuperUserUpserted
+func shouldUpsertUsers(dc api.CassandraDatacenter) bool {
+	lastCreated := dc.Status.UsersUpserted
 	return time.Now().After(lastCreated.Add(time.Minute * 4))
 }
 
-func (rc *ReconciliationContext) CreateSuperuser() result.ReconcileResult {
-	if rc.Datacenter.Spec.Stopped {
-		rc.ReqLogger.Info("cluster is stopped, skipping CreateSuperuser")
-		return result.Continue()
-	}
+func (rc *ReconciliationContext) createUser(user api.CassandraUser) error {
+	dc := rc.Datacenter
+	namespace := dc.ObjectMeta.Namespace
 
-	//Skip upsert if already did so recently
-	if !shouldUpsertSuperUser(*rc.Datacenter) {
-		rc.ReqLogger.Info(fmt.Sprintf("The CQL superuser was last upserted at %v, skipping upsert", rc.Datacenter.Status.SuperUserUpserted))
-		return result.Continue()
+	namespacedName := types.NamespacedName{
+		Name:      user.SecretName,
+		Namespace: namespace,
 	}
 
-	rc.ReqLogger.Info("reconcile_racks::CreateSuperuser")
+	secret, err := rc.retrieveSecret(namespacedName)
 
-	// Get the secret
-	secret, err := rc.retrieveSuperuserSecret()
 	if err != nil {
-		rc.ReqLogger.Error(err, "error retrieving SuperuserSecret for CassandraDatacenter.")
-		return result.Error(err)
+		return err
 	}
 
 	// We will call mgmt API on the first pod
-
 	pod := rc.dcPods[0]
 
 	err = rc.NodeMgmtClient.CallCreateRoleEndpoint(
 		pod,
 		string(secret.Data["username"]),
-		string(secret.Data["password"]))
-	if err != nil {
-		rc.ReqLogger.Error(err, "error creating superuser")
-		return result.Error(err)
+		string(secret.Data["password"]),
+		user.Superuser)
+
+	return err
+}
+
+func (rc *ReconciliationContext) CreateUsers() result.ReconcileResult {
+	dc := rc.Datacenter
+
+	if dc.Spec.Stopped {
+		rc.ReqLogger.Info("cluster is stopped, skipping CreateUser")
+		return result.Continue()
+	}
+
+	//Skip upsert if already did so recently
+	if !shouldUpsertUsers(*dc) {
+		rc.ReqLogger.Info(fmt.Sprintf("The users were last upserted at %v, skipping upsert", 
+			dc.Status.UsersUpserted))
```

Comment:
Better to keep the logging structured...

```suggestion
		rc.ReqLogger.Info("users upserted recently, skipping upsert", 
			"lastUpsert", dc.Status.UsersUpserted)
```